packages/components/src/DataList/DataList.css

```diff
@@ -35,6 +35,30 @@
   border-bottom: var(--border-base) solid var(--color-border);
 }
 
+.selectable {
+  display: grid;
+  column-gap: var(--space-small);
+  align-items: flex-start;
+  grid-template-columns: max-content minmax(0px, auto);
+
+  @media (--medium-screens-and-up) {
+    align-items: center;
+  }
+}
+
+.listItem .selectable > :first-child {
+  visibility: visible;
+  @media (--medium-screens-and-up) {
+    visibility: hidden;
+  }
+}
+
+.listItem:hover .selectable > :first-child,
+.listItem:focus .selectable > :first-child,
```

Comment:
`listItem` isn't focusable yet so this won't do anything. If this is meant to show the checkbox the moment you focus on the checkbox, `focus-within` is what you're looking for